src/DataDepsGenerators.jl

```diff
@@ -190,4 +282,10 @@ function mainpage_url(repo::DataRepo, dataname)
     getpage(url), url
 end
 
+struct GeneratorError <: Exception
+    repo::DataDepsGenerators.DataRepo
+end
```

Comment:
```
struct GeneratorError{T<:DataRepo} <: Exception
    repo::T
end
```

Don't put abstract types on fields, if it can be avoided, without making the code too ugly.
https://docs.julialang.org/en/stable/manual/performance-tips/#Avoid-fields-with-abstract-type-1

it isn't a major performance issue here, but it is good practice.

Reply:
Maybe give this a `message::String` field,
with default value `""`,
(so add the constructor `GeneratorError(repo)=GeneratorError(repo, "")`

And add that to the message to be shown during `showerror`.

Reply:
sorted now